src/PrettyPrinting.jl

```diff
@@ -1592,7 +1592,14 @@ function _write_line(io::IOCustom, str::AbstractString)
   end
   #@show spaceleft
   firstlen = min(spaceleft, length(str))
-  firststr = str[1:firstlen]
+  # make an iterator over valid indices
+  firstiter = Base.Iterators.take(eachindex(str), firstlen)
```

Comment:
I don't think this will handle combiners correctly (as often used for accents, e.g. in `í` unless normalized to latin).

Reply:
Actually if I enter this via copy&paste on my mac it works as it is normalized, but if I do `"i\acute<TAB>"` I get it:
```
julia> s = "í"
"í"

julia> collect(eachindex(s))
1-element Vector{Int64}:
 1

julia> s = "í"  # looks the same but isn't the same on a byte level
"í"

julia> collect(eachindex(s))
2-element Vector{Int64}:
 1
 2
```

Reply:
I think this can be circumvented by https://docs.julialang.org/en/v1/stdlib/Unicode/#Unicode.normalize

Reply:
For tests might be best to store the "bad" characters as byte sequences so we are sure that we are testing the right ones
```
julia> s = "í"
"í"

julia> Vector{UInt8}(s)
2-element Vector{UInt8}:
 0xc3
 0xad

julia> String(Vector{UInt8}(s))
"í"
```

Reply:
These are the characters that made me notice the issue:
```
julia> Vector{UInt8}("∧")
3-element Vector{UInt8}:
 0xe2
 0x88
 0xa7

julia> Vector{UInt8}("…")
3-element Vector{UInt8}:
 0xe2
 0x80
 0xa6
```

The latter e.g. with `orbits(symmetric_group(50))` (works for smaller values than 21 for me -- may depend on your terminal settings)

Reply:
This should be fixed by normalizing everything before use, right ?